Zend/zend_API.c

```diff
@@ -4281,6 +4281,19 @@ ZEND_API zend_bool zend_is_iterable(zval *iterable) /* {{{ */
 }
 /* }}} */
 
+ZEND_API zend_bool zend_is_countable(zval *countable) /* {{{ */
```

Comment:
If possible we should make this `zval const *`. At a glance it depends if `instanceof_function` and related calls are const or not.

Reply:
By convention, zvals are never passed by const pointer.

Reply:
So it should stay as it is?

Reply:
Can we please move this discussion elsewhere (i.e. php.internals) and focus on this patch? We currently have two discussions both unrelated to the patch itself. Thanks. 👍

Reply:
@Majkl578 You misunderstand. The const discussion *is* about the patch.

Reply:
> By convention, zvals are never passed by const pointer.

This really sounds like a global issue for all zvals, not just this patch. Trying to mitigate this convention is out of scope here.

Reply:
This is my opinion.: it appears `instanceof_function` takes its parameters by pointer to const and `Z_TYPE_P` should be const correct as well. This function should take its parameter by pointer to const because we have verified here, today, that it is const correct. It is easier to do this now than later. No other functions need to be adjusted for *this* function to work with const parameters. As such I believe this is in scope for this PR. If other functions would have to be adjust for it to work then it would be out of scope.

Reply:
@morrisonlevi To avoid this discution on _to be const or not to be_, should I move this `switch` to the function declaration, in `type.c` like #2206?

Also, may I ask you something? What are the difference between have the function here in `Zend_api.c` or there in `type.c`?

Reply:
Having it in the API means it can be called from other places in the C code. I think it belongs there - I would not move it inline like the other PR.